src/main/kotlin/Setup.kt

```diff
@@ -0,0 +1,58 @@
+import com.mongodb.client.model.CreateCollectionOptions
+import com.mongodb.kotlin.client.coroutine.MongoClient
+import com.mongodb.kotlin.client.coroutine.MongoDatabase
+import kotlinx.coroutines.runBlocking
+import java.util.*
+
+fun main() {
+    val databaseName = "sample_restaurants"
+    val database = setupConnection(databaseName = databaseName, "MONGODB_URI")
+    runBlocking {
+        listAllCollection(database = database)
+
+        createCollection(database = database)
+
+        listAllCollection(database = database)
+
+        dropCollection(database = database)
+
+        listAllCollection(database = database)
+    }
+}
+
+fun setupConnection(
+    databaseName: String = "sample_restaurants",
+    connectionEnvVariable: String = "MONGODB_URI"
+):
+        MongoDatabase {
+    val connectString = if (System.getenv(connectionEnvVariable) != null) {
+        System.getenv(connectionEnvVariable)
+    } else {
+        "mongodb+srv://<username>:<enter your password>@cluster0.sq3aiau.mongodb" +
+                ".net/?retryWrites=true&w=majority"
+    }
+    val client = MongoClient.create(connectionString = connectString)
+    return client.getDatabase(databaseName = databaseName)
+}
+
+suspend fun listAllCollection(database: MongoDatabase) {
+
+    print("Collection in this database are ---------------> ")
+    database.listCollectionNames().collect { print(" $it") }
+    println()
+}
+
+suspend fun createCollection(database: MongoDatabase) {
```

Comment:
From @johnlpage - Why createCollection? and what on earth is maxDocuments? (Ah I see but the collection must be capped to use it)

Reply:
@johnlpage: Just to demo how to create collection :) Are you suggesting something else or anything wrong with this approach?

Reply:
I'm suggesting thart no-one should be calling createCollection when using MongoDB unless they have some unusual need (*like time series) so don't do it.